content/posts/0.19-release.md

```diff
@@ -0,0 +1,98 @@
+---
+title: "Pyodide 0.19.0 release"
+date: 2022-01-07
+draft: true
+tags: ["internals"]
+author: "Hood Chatham and Roman Yurchak"
+showToc: true
+TocOpen: false
+draft: false
+hidemeta: false
+comments: false
+# description: ""
+disableHLJS: true # to disable highlightjs
+disableShare: false
+hideSummary: false
+searchHidden: true
+ShowReadingTime: true
+ShowBreadCrumbs: true
+ShowPostNavLinks: true
+---
+
+Pyodide 0.19 is a major release focused on improving the build system, updating
+core scientific dependencies, and improving the workflow for loading Python
+code.
+
+### Removal of function pointer cast emulation
+
+The Python C extensions occasionally use function points casts, which is an
+undefined behavior in C specification. However in practice, all native C
+compilers produce code that works fine.  This is not the case with Emscripten
+when building for WebAssembly, which will fails on function pointer casts.  As
+a workaround we have been using Emscripten's `-s
+EMULATE_FUNCTION_POINTER_CASTS` flag, which emits code to emulate function
+pointer casts at runtime. This has however a significant performance and size
+overhead.  
+
+In version 0.19 we were able to remove the function cast emulation by using
+Javascript method call trampolines in a few places where the function pointer
+casts are happening.
+This allowed to increase the maximum supported redursion depth to the
+default value of 1000, decrease the compiled code size by ~25% and make Python
+run faster by 10 to 20%.  For more details about this work, see our post on
+[Function Pointer Cast Handling in
+Pyodide](https://blog.pyodide.org/posts/function-pointer-cast-handling/).
+
+### Scipy Upgrade
+
+The initial version of SciPy added to Pyodide was 0.17.1 from 2016. Since then
+we had major difficulties in updating it, due Scipy being quite large,
+difficult to build, and due to the use of Fortran which has no reliable
+compiler supporting WebAssembly. The inability to update SciPy, in turn
+prevented the update of other scientific Python packages, such as scikit-learn,
+scikit-image and statsmodels.
+
+In this version, after a lot work, Scipy was finally updated in
+https://github.com/pyodide/pyodide/pull/2065 as well packages that dependended
+on it.  There are still known issues with Scipy packaged in Pyodide, and components don't work. However,
+this update is a major milestone in getting us closer to the development
+version of Scipy and allowing to contribute fixes upsteam. This also allowed 
```

Comment:
The two sections above are ready for review.

The following ones still need work.